pubsub/gcp/pubsub/pubsub.go

```diff
@@ -354,9 +416,25 @@ func (g *GCPPubSub) getTopic(topic string) *gcppubsub.Topic {
 }
 
 func (g *GCPPubSub) ensureSubscription(parentCtx context.Context, subscription string, topic string) error {
-	err := g.ensureTopic(parentCtx, topic)
-	if err != nil {
-		return err
+	g.lock.RLock()
+	_, TExists := g.topicCache[topic]
+	_, DExists := g.topicCache[g.metadata.DeadLetterTopic]
```

Comment:
We don't capitalise local variable names. It is also convention to use `ok` as the name for the bool returned when checking a map.

```suggestion
	_, topicOK := g.topicCache[topic]
	_, dlTopicOK := g.topicCache[g.metadata.DeadLetterTopic]
```